esmvaltool/cmorizers/data/formatters/datasets/iap.py

```diff
@@ -0,0 +1,193 @@
+# pylint: disable=unused-argument
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-function-args
+# pylint: disable=R0917
+# pylint: disable=E1121
+# flake8: noqa
+"""ESMValTool CMORizer for IAP data.
+
+Tier
+   Tier 2: other freely-available dataset.
+
+Source
+   IAPv4.2: "http://www.ocean.iap.ac.cn/ftp/cheng/"
+            "IAPv4.2_IAP_Temperature_gridded_1month_netcdf/Monthly/"
+
+Last access: 20250220
+
+Download and processing instructions
+   All handled by the script (download only if local data are missing)
+
+   Alternatively, download and unzip the following files:
+     Temperature_IAPv4.2_gridded_data_1940_1949.zip
+     Temperature_IAPv4.2_gridded_data_1950_1959.zip
+     Temperature_IAPv4.2_gridded_data_1960_1969.zip
+     Temperature_IAPv4.2_gridded_data_1970_1979.zip
+     Temperature_IAPv4.2_gridded_data_1980_1989.zip
+     Temperature_IAPv4.2_gridded_data_1990_1999.zip
+     Temperature_IAPv4.2_gridded_data_2000_2009.zip
+     Temperature_IAPv4.2_gridded_data_2010_2019.zip
+     Temperature_IAPv4.2_gridded_data_2020_2023.zip
+"""
+
+import logging
+import os
+import warnings
+from warnings import catch_warnings
+from datetime import datetime
+from dateutil import relativedelta
+
+import iris
+import cf_units
+import numpy as np
+
+from esmvaltool.cmorizers.data.utilities import (
+    fix_coords,
+    fix_var_metadata,
+    save_variable,
+    set_global_atts,
+)
+
+logger = logging.getLogger(__name__)
+
+try:
+    iris.FUTURE.date_microseconds = True
+    iris.FUTURE.save_split_attrs = True
+except AttributeError as e:
+    # Handle cases where FUTURE or the attributes don't exist
+    logger.warning("AttributeError: %s", e)
+except (TypeError, ValueError) as e:
+    # Handle specific errors if these might occur
+    logger.warning("TypeError or ValueError: %s", e)
+except BaseException as e:
+    # Fallback for rare or unknown issues, but avoid catching Exception
+    logger.warning("An unexpected error occurred: %s", e)
+
+def collect_files(in_dir, cfg, start_date, end_date):
+    """ Create list of files path to be processed."""
+    file_list = []
+
+    if start_date is None:
+        start_date = datetime(year=1940, month=1, day=1)
+    if end_date is None:
+        end_date = datetime(year=2024, month=12, day=31)
+
+    loop_date = start_date
+
+    while loop_date <= end_date:
+        fname = (
+            f"IAPv4_Temp_monthly_1_6000m_year_{loop_date.year}"
+            f"_month_{loop_date.month:02d}.nc"
+        )
+        in_file = os.path.join(in_dir, fname)
+        file_list.append(in_file)
+        loop_date += relativedelta.relativedelta(months=1)
+
+    return file_list
+
+
+def process_data(cube, reference_year):
+    """ Process raw data. Convert to Kelvin and add time dimension.
+        Concatenate the cubes and return the new cube.
+    """
+    # Convert temperature from Celsius to Kelvin and add time dimension
```

Comment:
```suggestion
    # Add time dimension
```
not converting to Kelvin